components/camel-opentracing/src/main/java/org/apache/camel/opentracing/decorators/AbstractSpanDecorator.java

```diff
@@ -33,12 +33,40 @@
 
     @Override
     public String getOperationName(Exchange exchange, Endpoint endpoint) {
-        return endpoint.getEndpointUri();
+        // OpenTracing aims to use low cardinality operation names. Ideally a specific
+        // span decorator should be defined for all relevant Camel components that
+        // identify a meaningful operation name
+        return URI.create(endpoint.getEndpointUri()).getScheme();
+    }
+
+    /**
+     * This method removes the scheme, any leading slash characters and
+     * options from the supplied URI. This is intended to extract a meaningful
+     * name from the URI that can be used in situations, such as the operation
+     * name.
+     *
+     * @param endpoint The endpoint
+     * @return The stripped value from the URI
+     */
+    public static String stripSchemeAndOptions(Endpoint endpoint) {
+        int start = endpoint.getEndpointUri().indexOf(':');
+        start++;
+        // Remove any leading '/'
+        while (endpoint.getEndpointUri().charAt(start) == '/') {
+            start++;
+        }
+        int end = endpoint.getEndpointUri().indexOf('?');
+        return end == -1 ? endpoint.getEndpointUri().substring(start)
+                : endpoint.getEndpointUri().substring(start, end);
     }
 
     @Override
     public void pre(Span span, Exchange exchange, Endpoint endpoint) {
         span.setTag(Tags.COMPONENT.getKey(), CAMEL_COMPONENT + URI.create(endpoint.getEndpointUri()).getScheme());
+
+        // Including the endpoint URI provides access to any options that may have been provided, for
+        // subsequent analysis
+        span.setTag("camel.uri", endpoint.getEndpointUri());
```

Comment:
I wonder if we should mask sensitive information such as passwords, which Camel do with

org.apache.camel.util.URISupport#sanitizeUri

which it does by default in its own toString